src/main/java/org/folio/edge/sip2/domain/messages/requests/PatronInformation.java

```diff
@@ -0,0 +1,217 @@
+  private final String patronIdentifier;
+  /** The password used by the terminal. */
+  private final String terminalPassword;
+  /** The password used by the patron. */
+  private final String patronPassword;
+  /** The number of the first item to be returned. */
+  private final Integer startItem;
+  /** The number of the last item to be returned. */
+  private final Integer endItem;
+
+  /**
+   * Construct a {@code PatronInformation} based on a
+   * {@code PatronInformationBuilder} object.
+   * @param builder The basis for creating the patron information.
+   */
+  private PatronInformation(PatronInformationBuilder builder) {
+    this.language = builder.language;
+    this.transactionDate = builder.transactionDate;
+    this.summary = builder.summary;
+    this.institutionId = builder.institutionId;
+    this.patronIdentifier = builder.patronIdentifier;
+    this.terminalPassword = builder.terminalPassword;
+    this.patronPassword = builder.patronPassword;
+    this.startItem = builder.startItem;
+    this.endItem = builder.endItem;
+  }
+
+  /**
+   * Returns a builder used to construct a {@code PatronInformation}.
+   * @return A patron information builder.
+   */
+  public static PatronInformationBuilder builder() {
+    return new PatronInformationBuilder();
+  }
+
+  public Language getLanguage() {
+    return language;
+  }
+
+  public ZonedDateTime getTransactionDate() {
+    return transactionDate;
+  }
+
+  public Summary getSummary() {
+    return summary;
+  }
+
+  public String getInstitutionId() {
+    return institutionId;
+  }
+
+  public String getPatronIdentifier() {
+    return patronIdentifier;
+  }
+
+  public String getTerminalPassword() {
+    return terminalPassword;
+  }
+
+  public String getPatronPassword() {
+    return patronPassword;
+  }
+
+  public Integer getStartItem() {
+    return startItem;
+  }
+
+  public Integer getEndItem() {
+    return endItem;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(endItem, institutionId, language, patronIdentifier,
+        patronPassword, startItem, summary, terminalPassword, transactionDate);
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) {
+      return true;
+    }
+    if (obj == null) {
+      return false;
+    }
+    if (!(obj instanceof PatronInformation)) {
+      return false;
+    }
+    PatronInformation other = (PatronInformation) obj;
+    return Objects.equals(endItem, other.endItem)
+        && Objects.equals(institutionId, other.institutionId)
+        && language == other.language
+        && Objects.equals(patronIdentifier, other.patronIdentifier)
+        && Objects.equals(patronPassword, other.patronPassword)
+        && Objects.equals(startItem, other.startItem)
+        && summary == other.summary
+        && Objects.equals(terminalPassword, other.terminalPassword)
+        && Objects.equals(transactionDate, other.transactionDate);
+  }
+
+  @Override
+  public String toString() {
```

Comment:
Typically, `toString` is used for logging. I've never seen reflection used for `toString()`. I don't think the overhead and code complexity would be really worth it for such a simple method. But, if you find a super cool way to do it via reflection it would be easy to refactor. 😄

Reply:
I just thought that we're doing this for every class so if we just come up with a way of doing it via reflection once then we only need to copy and paste it into all the classes or at the very least have it in a parent class so that all could inherit.

Reply:
Yeah, that is reasonable. I found an tostring implementation using reflection that apache provides in one of their commons packages that we could use: https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/ReflectionToStringBuilder.html

I still think it might be more effort to ensure that all objects are printed properly, though that class does provide some mechanism for customization. Eclipse autogenerates the `toString` method, so it is not a big deal, at least to me, to just generate it for each class.